File: RedSheet.DbEntities/Models/vBackgroundEvent.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vBackgroundEvents", Schema = "dbo")]
	[Serializable]
	public partial class vBackgroundEvent  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int BackgroundEventId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Description { get; set; } 

	public Nullable<System.DateTime> EndDate { get; set; } 
 
	[Required]
	public bool IsEvent { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectBackgroundId { get; set; } 
 
	[Required]
	public System.DateTime StartDate { get; set; } 
 
	[Required] 
	[MaxLength(1012)]
	public string Title { get; set; } 

	public vBackgroundEvent()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-background-event.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vBackgroundEvent {
    constructor(vBackgroundEvent?: vBackgroundEvent )  {
        let properties = [ "backgroundEventId", "description", "endDate", "isEvent", "projectBackgroundId", "startDate", "title",];
        for (let property of properties)
            if (vBackgroundEvent && vBackgroundEvent[property])
                this[property] = vBackgroundEvent[property];
    }
 
	backgroundEventId : number =   0 ;
 
    @required()
    @maxLength(1000)
	description : string =   undefined;
 
	endDate : Date =   undefined;
 
	isEvent : boolean = false ;
 
    @range(1,2147483647)
	projectBackgroundId : number =   undefined;
 
    @required()
	startDate : Date =   undefined;
 
    @required()
    @maxLength(1012)
	title : string =   undefined;


}
